src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -257,11 +283,28 @@ def create_key(self, arg0, arg1=None, arg2=None, names='i,j,k'):
             a = K(v[0])
             b = K(v[1])
 
-        # QuaternionAlgebra(K, a, b)
         else:
+            # QuaternionAlgebra(K, finite_places, archimedian_places)
             K = arg0
-            a = K(arg1)
-            b = K(arg2)
+            if isinstance(arg1, list) and isinstance(arg2, list):
+                if K == QQ:
+                    D = ZZ.ideal_monoid().prod(arg1).gen()
+                    a, b = hilbert_conductor_inverse(D)
+                    a = Rational(a)
+                    b = Rational(b)
```

Comment:
I think it would be nice to redesign it so that the one element list with the correct invariant at infinity as well as the empty list (signaling the constructor to solve for the invariant at infinity on its own) are precisely the valid second arguments.
Also a check/error catch for the content of arg1 should be included here as well.

Reply:
That's a good question, I'd say there's arguments both for and against forcing uniqueness. I would not force it, but that means you should make sure to not take one prime factor twice (and similarly for the number field case; Pari throws an appropriate error, but since all Pari errors are of the same type `PariError` you probably want to enforce this in advance to still correctly catch the invariant error), e.g. by converting the list of primes into a set beforehand.